File: UI/app/src/main/java/com/rikkei/training/ui/Adapter.java
package com.rikkei.training.ui;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class Adapter extends RecyclerView.Adapter<Adapter.ImageHolder> {
    private LayoutInflater inflater;
    private List<Image> data;

    public Adapter(Context context) {
        inflater = LayoutInflater.from(context);
    }

    public void setData(List<Image> data) {
        this.data = data;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ImageHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = inflater.inflate(R.layout.item,parent,false);

        return new ImageHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ImageHolder holder, int position) {
        Image item = data.get(position);
        holder.bindData(item);

    }

    @Override
    public int getItemCount() {
        return data==null?0:data.size();
    }

    public class ImageHolder extends RecyclerView.ViewHolder{
        private ImageView img;
        public ImageHolder(@NonNull View itemView) {
            super(itemView);
            img = itemView.findViewById(R.id.img_item);
        }
        public void bindData(Image item){
            img.setImageResource(item.getImg());
        }
    }
}

File: UIKotlin/app/src/main/java/com/rikkei/training/uikotlin/Adapter.kt
package com.rikkei.training.uikotlin

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.rikkei.training.uikotlin.model.Image
import kotlinx.android.synthetic.main.item.view.*

class Adapter(private var data: ArrayList<Image>, private val context: Context) :
    RecyclerView.Adapter<Adapter.ImageHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ImageHolder {
        return ImageHolder(LayoutInflater.from(context).inflate(R.layout.item, parent, false))
    }

    override fun getItemCount(): Int {
        return data.size
    }

    override fun onBindViewHolder(holder: ImageHolder, position: Int) {
        holder.onBind(data[position])
    }


    class ImageHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val itemViewType = itemView.img_item
        fun onBind(item: Image){
            itemViewType.setImageResource(item.img!!)
        }
    }
}